### ISSUE TITLE
How to set a memory limit for a sandboxed processor?

### ISSUE THREAD

**Issue by ppedziwiatr** [taskforcesh/bullmq#1555]

**Comment by manast** [comment]
Currently they are implemented as spawn processes, so there is no such thing as memory configuration AFAIK.

**Comment by ppedziwiatr** [comment]
Thanks for the answer. What is the advantage of using a spawned process (in comparison to worker threads)?

**Comment by ppedziwiatr** [comment]
Ok, if I understand properly, they are implemented as forks, (not as spawned processes) -https://github.com/taskforcesh/bullmq/blob/master/src/classes/child-pool.ts#L114

Which is event better, because I believe that each fork has a new V8 instance created = better isolation from the "host" envrionment.

And if that's really the case - it should be possible to set the memory limit per forked process - with sth like:
```
require('child_process').fork("newProcess.js", { 
        execArgv: ['--max-old-space-size=128']
    });)
```    
https://stackoverflow.com/a/37596117

Would you consider adding the max available memory setting in the Worker configuration options?

**Comment by manast** [comment]
Yeah well, a fork is also a spawn, the only difference is that it creates a new V8 for you, but it is still a new process so there is no practical difference (with spawn you will need to call the node cmd line tool instead).
But sure, we can add support for setting custom flags to child processes, the tricky part is how would such an API look like so that it is consistent with current design.

**Comment by ppedziwiatr** [comment]
For me personally - the simplest solution (assuming, that this is a more of "advanced" feature) sth like this:
```js
new Worker(updateQueueName, updateProcessor, {
    concurrency: workersConfig.update,
    metrics: {
      maxDataPoints: MetricsTime.ONE_WEEK,
    },
    sandboxExecArgv: ['--max-old-space-size=128', '--foo=bar']
  });
```

would be awesome!

**Comment by ppedziwiatr** [comment]
Hey, any updates? :-)

**Comment by ppedziwiatr** [comment]
Hey - just asking while reviewing our internal issues ;-) - is there any timeline for implementing this feature?

**Comment by ppedziwiatr** [comment]
Hey, do you plan to implement this feature?

**Comment by manast** [comment]
It does not seem to be a lot of interest in this feature. However if someone creates a PR with a clean and simple implementation I do not have anything against merging it.

**Comment by opexu** [comment]
i need this feature too
